server/src/main/scala/net/wiringbits/actions/CreateUserAction.scala

```diff
@@ -4,26 +4,34 @@ import net.wiringbits.api.models.CreateUser
 import net.wiringbits.apis.ReCaptchaApi
 import net.wiringbits.config.UserTokensConfig
 import net.wiringbits.repositories
-import net.wiringbits.repositories.UsersRepository
-import net.wiringbits.repositories.models.User
+import net.wiringbits.repositories.models.UserTokenType
 import net.wiringbits.util.{EmailsHelper, TokenGenerator, TokensHelper}
 import net.wiringbits.validations.{ValidateCaptcha, ValidateEmailIsAvailable}
+import org.foo.generated.customtypes.{TypoOffsetDateTime, TypoUUID, TypoUnknownCitext}
+import org.foo.generated.public.user_logs.{UserLogsId, UserLogsRepoImpl, UserLogsRow}
+import org.foo.generated.public.user_tokens.{UserTokensId, UserTokensRepoImpl, UserTokensRow}
+import org.foo.generated.public.users.{UsersId, UsersRepoImpl, UsersRow}
 import org.mindrot.jbcrypt.BCrypt
+import play.api.db.Database
 
-import java.time.Instant
+import java.sql.Connection
+import java.time.temporal.ChronoUnit
+import java.time.{Clock, OffsetDateTime, ZoneOffset}
 import java.util.UUID
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class CreateUserAction @Inject() (
-    usersRepository: UsersRepository,
     reCaptchaApi: ReCaptchaApi,
     tokenGenerator: TokenGenerator,
     userTokensConfig: UserTokensConfig,
-    emailsHelper: EmailsHelper
+    emailsHelper: EmailsHelper,
+    database: Database
 )(implicit
+    clock: Clock,
     ec: ExecutionContext
 ) {
+  given db: Connection = database.getConnection()
```

Comment:
This will hold a connection for the `CreateUserAction` lifetime which is not ideal, the connection must be released as soon as we are done with the operation